src/main/java/Duke.java

```diff
@@ -1,10 +1,178 @@
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+    private static final ArrayList<Task> tasks = new ArrayList<>();
+
     public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String line = "________________________________________";
+        String curCommand = "";
+
+        readFromFile();
+        writeIntoFile();
+
+        System.out.println(logo);
+        System.out.println(line);
+        System.out.println("Hello! I'm Duke\nWhat can I do for you?");
+        System.out.println(line);
+
+        while (!curCommand.equals("bye")) {
+            curCommand = input.nextLine();
+            System.out.println(line);
+            handleCommand(curCommand);
+            System.out.println(line);
+        }
     }
+
+    // Handle command
+    public static void handleCommand(String curCommand) {
+        String[] commands = curCommand.split(" ", 2);
+        if (commands.length > 1) {
+            handleMultiCommands(commands);
+        } else {
+            handleSingleCommand(curCommand);
+        }
+    }
+
+    // Handle multiple commands
+    public static void handleMultiCommands(String[] commands) {
+        switch (commands[0]) {
+        case "done":
+            int doneIndex = getPositiveNumeric(commands[1]);
+            if (doneIndex > 0 && doneIndex <= tasks.size()) {
+                tasks.get(doneIndex - 1).setDone(true);
+            }
+            System.out.println(doneIndex > 0 && doneIndex <= tasks.size()
+                    ? "Nice! I've marked this task as done:\n " + MessageManager.taskToString(tasks.get(doneIndex - 1))
+                    : "Formatting error");
+            writeIntoFile();
+            break;
+        case "todo":
+            tasks.add(new Task(commands[1], "", 'T', false));
+            MessageManager.printNewTask(tasks.get(tasks.size() - 1), tasks.size());
+            writeIntoFile();
+            break;
+        case "deadline":
+            String[] deadin = commands[1].split(" /by ", 2);
+            if (deadin.length > 1) {
+                tasks.add(new Task(deadin[0], deadin[1], 'D', false));
```

Comment:
Avoid doing class initialization in the argument of another function. Store the new class in a variable and then pass that variable as the argument to the function